Update ctags.readme.txt (#4360)
gap-system/gap: etc/ctags.readme.txt

> Run

Reviewer: Maybe add an initial sentence/paragraph to explain what this is all about?

  ctags --version
to see whether you are running Exuberant Ctags or Universal Ctags.
Append the content of the file ctags_for_gap to the file

>   ~/.ctags

Reviewer: Note that none of this is necessary if one just wants to use ctags in GAP, at least with excuberant ctags: we already have a .ctags file (well, symlink) in the gap root dir!

Author: Which of course does not work with Universal Ctags *sigh*...
Anyways, I'll add a comment at the top of the file.

if you are running Exuberant Ctags. If you are running Universal Ctags then
append the content of the file ctags_for_gap to the file
  $XDG_CONFIG_HOME/ctags/gap.ctags
or
  $HOME/.config/ctags/gap.ctags
if $XDG_CONFIG_HOME is not defined.
Then run
  ctags --sort=no lib/*.g lib/*.gi lib/*.gd src/*
in the GAP installation directory. This creates a file
  tags

> Start your editor in that directory and then do (for vim e.g.):

Reviewer: This is phrased weirdly. Maybe:
```suggestion
Many editors support ctags either natively or via a plugin (e.g. vim, emacs, BBEdit, ...). For example, to use it in vim, start it in that directory and then do:
```

This file describes how to set up Ctags to work with GAP source files.
Ctags is a tool that can be used to generate an index file of names found in
source files. This index file can then be used by editors to jump to the
definition of any indexed name.

Note that if you have Exuberant Ctags installed (if in doubt, run `ctags --version`)
and only want to generate tags for the gap-system/gap repository, then you do

> not need set up anything. The symlink `.ctags` in the root folder already

Reviewer: ```suggestion
not need set up anything. The symlink `.ctags` in the GAP root folder already
```

tells ctags all it needs to know, and you can create and update the tags by
running:
  make tags
If you want to use ctags in another GAP project or have Universal Ctags
installed, read on.

Run
  ctags --version
to see whether you are running Exuberant Ctags or Universal Ctags.
Append the content of the file ctags_for_gap to the file
  ~/.ctags

> if you are running Exuberant Ctags. If you are running Universal Ctags then

Reviewer 2: Make the two cases clearer here. Maybe use paragraphs?

Reviewer: Bonus idea: convert the file to markdown and call it `README.ctags.md`; then use four space indent and surrounding empty lines to turn this into code blocks.

But of course this goes way beyond the original scope of this PR. so please do not feel compelled to do this unless you really want to.

append the content of the file ctags_for_gap to the file
  $XDG_CONFIG_HOME/ctags/gap.ctags
or
  $HOME/.config/ctags/gap.ctags
if $XDG_CONFIG_HOME is not defined.
Then run
  ctags --recurse lib/ src/
in the GAP installation directory. If you are running ctags in another project,
you may have to do
  ctags --recurse gap/ src/
instead. This creates a file
  tags
Many editors support ctags either natively or via a plugin (e.g. vim, emacs,
BBEdit, ...). For example, to use it in vim, start it in that directory and
then do:
  :tag DerivedSubgroup
(say). You can also use
  :ts DerivedSubgroup
to get an overview over all tags found (which is useful since it shows
for example all method installations).

> For instructions on how to create tags automatically via git-hooks, for example

Reviewer: ```suggestion
For instructions on how to regenerate tags automatically via git-hooks, for example
```
